spec/abci/abci++_app_requirements.md

```diff
@@ -188,6 +192,34 @@ Likewise, `ExtendVote` can also be non-deterministic:
 * *w<sup>r</sup><sub>p</sub> = w<sup>r</sup><sub>q</sub> &#8655;
   e<sup>r</sup><sub>p</sub> = e<sup>r</sup><sub>q</sub>*
 
+### Mempool Connection Requirements
+
+Before defining our notation, we need to introduce a requirement for `CheckTx` to always
+yield the same result while *p*'s Application is at a given height.
+
+* Requirement 13 [`CheckTx`, same-height immutability]: For any correct process *p*,
+  any transaction *tx*, and any height *h<sub>p</sub>*,
+  `ResponseCheckTx` returns the same result code while *p* is at *h<sub>p</sub>*.
+
+Requirement 13 allows us to use heights rather than time in our notation.
+Let *CheckTxCode<sub>tx,p,h</sub>* denote the result  code returned by *p*'s Application, via `ResponseCheckTx`,
+to CometBFT's `RequestCheckTx` call occurring while the Application is at height *h*
+and having transaction *tx* as parameter.
+Let predicate *OK(CheckTxCode)* denote whether *CheckTxCode* is `SUCCESS`.
+
+* Requirement 14 [`CheckTx`, eventual non-oscillation]: For any correct process *p*
+  and any transaction *tx*, there exists a height *h<sub>p,stable</sub>*
```

Comment:
Why the height is associated to a particular process? The advantage of using heights instead of time is to make the property global. In fact, since applications are deterministic, the outputs produced by different applications at the same height (which is a global parameter) should be the same.

Reply:
Yes, but I didn't want to make this requirement global. This is aimed at the app devs. I want the requirements addressed to them to be as imple and understandable as possible.
Then, assuming they respect those, we can deduce that a global property emerges (thanks to consensus agreement)... but we shouldn't confuse app devs by introducing all that "complication" here